docs/tutorials/typescript/nextjs/index.md

```diff
@@ -4,273 +4,608 @@ sidebar_position: 2
+              ERROR:   ( <button onClick={buyProduct} className={errorStyle}>Error! Click to Retry </button> ),
+            }[state]
+          }
+        </div>
+      </div>
+      <div className="mt-4 flex items-center justify-between text-base font-medium text-gray-900 space-x-8">
+        <h3>{product.name}</h3>
+        <p>{product.price}</p>
+      </div>
+    </div>
+  );
+};
+```
+<!--SNIPEND-->
+
+In this component, the `buyProduct` function makes the call to start the Temporal Workflow by making a request to the API route you defined.
+The component renders the product and displays a button for the customer to click. Based on the order state,
+the button is replaced with a confirmation message or a button that lets the customer try again if there's an error.
+
+Now add a `ProductList` component with the following code:
+
+<!--SNIPSTART typescript-next-oneclick-page-productlist -->
+[app/page.tsx](https://github.com/temporalio/nextjs-temporal-one-click-template/blob/next-v2/app/page.tsx)
+```tsx
+const ProductList: React.FC = () => {
+  return (
+    <div className="bg-white">
+      <div className="max-w-2xl mx-auto py-16 px-4 sm:py-24 sm:px-6 lg:max-w-7xl lg:px-8">
+        <div className="mt-6 grid grid-cols-1 gap-x-8 gap-y-8 sm:grid-cols-2 sm:gap-y-10 md:grid-cols-4">
+          {products.map((product) => (
+            <Product product={product} key={product.id} />
+          ))}
+        </div>
+      </div>
+    </div>
+  );
+};
+```
+<!--SNIPEND-->
+
+Finally, add the `Home` component to define the overall header and render the product list:
+
+<!--SNIPSTART typescript-next-oneclick-page-home -->
+[app/page.tsx](https://github.com/temporalio/nextjs-temporal-one-click-template/blob/next-v2/app/page.tsx)
+```tsx
+const Home: React.FC = () => {
+  return (
+    <div className="pt-8 pb-80 sm:pt-12 sm:pb-40 lg:pt-24 lg:pb-48">
+      <div className="relative max-w-7xl mx-auto px-4 sm:px-6 lg:px-8 sm:static">
+        <Head>
+          <title>Temporal + Next.js Example</title>
+          <link rel="icon" href="/favicon.ico" />
+        </Head>
+        <header className="relative overflow-hidden">
+          <div className="sm:max-w-lg">
+            <h1 className="text-4xl font font-extrabold tracking-tight text-gray-900 sm:text-6xl">
+              Temporal.io + Next.js One Click Purchase Demo
+            </h1>
+            <p className="mt-4 text-xl text-gray-500">
+              Click on the item to buy it now.
+            </p>
+          </div>
+        </header>
+        <ProductList />
+      </div>
+    </div>
+  );
+};
+
+export default Home;
+```
+<!--SNIPEND-->
 
-However, your Temporal Workers **must** be deployed in traditional "serverful" environments (e.g. with EC2, Digital Ocean or Render, not a serverless environment).
+Visit `http://localhost:3000` in your browser and you'll see two products. Click their buttons and you'll execute the Temporal Workflows.
```

Comment:
not sure if `http://localhost:3000` should be a link or not

Reply:
I think no, but I'm open to revisiting this on a larger scale as part of style. There are benefits for sure.